code/modules/projectiles/boxes_magazines/internal/_cylinder.dm

```diff
@@ -23,14 +28,7 @@
 		rotate()
 
 /obj/item/ammo_box/magazine/internal/cylinder/ammo_list(drop_list = FALSE)
-	var/list/L = list()
-	for(var/i=1 to stored_ammo.len)
-		var/obj/item/ammo_casing/bullet = stored_ammo[i]
-		if(bullet)
-			L.Add(bullet)
-			if(drop_list)//We have to maintain the list size, to emulate a cylinder
-				stored_ammo[i] = null
-	return L
+	return list_clear_nulls(stored_ammo.Copy())
```

Comment:
Is this tested? `list_clear_nulls` does not return the list it is modifying, it returns `TRUE` if it removed any nulls.

Reply:
Oh, I forgot for a hot minute this also affected guns. Thanks!